### PR TITLE
Update copied docs and pipeline docs

### PR DESCRIPTION
**Pull request by simitt**
* Update docs copied from beats
* Update pipeline related docs, after merging automatic pipeline registration elastic/apm-server#1258 .

screen shot 2018-08-17 at 10 02 01

### CODE REVIEW

File: docs/configuring-ingest.asciidoc [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        }
+    ]
+}
+------------------------------------------------------------------------------
+
+To add the pipeline in Elasticsearch, you would run:
+
+[source,shell]
+------------------------------------------------------------------------------
+curl -H 'Content-Type: application/json' -XPUT 'http://localhost:9200/_ingest/pipeline/test-pipeline' -d@pipeline.json
```
Comment (jalvz) [inline docs/configuring-ingest.asciidoc]: i think you miss an space: `-d @pipeline.json`

File: docs/configuring-ingest.asciidoc [hunk 2]
```diff
@@ -0,0 +1,87 @@
+// This file was copied over from libbeat and
+// then adapted to APM Server specific needs
+
+[[configuring-ingest-node]]
+== Parse data using ingest node pipelines
+
+When you use Elasticsearch for output, you can configure {beatname_uc} to use
```
Comment (jalvz) [inline docs/configuring-ingest.asciidoc]: Alternative suggestion:

With the Elasticsearch output you can configure the APM Server to use
{elasticsearch}/ingest.html[ingest node] to pre-process documents before indexing them. 
A pipeline defines processors that operate on your data. 
For example, you can register a pipeline that defines one processor that removes a field and another that renames other field.

File: docs/configuring-ingest.asciidoc [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+[float]
+=== Registering Pipelines in Elasticsearch
+For registering a pipeline in Elasticsearch you can either manually upload
+a pipeline definition or you can configure APM Server to register pipelines on startup.
+
+[[register-pipelines-apm-server]]
+[float]
+==== Register pipelines on APM Server startup
+Automatic pipeline registration requires `output.elasticsearch` to be enabled and configured,
+as well as having the required Elasticsearch Processor Plugins installed.
```
Comment (jalvz) [inline docs/configuring-ingest.asciidoc]: "... as having the required Elasticsearch plugins installed"

File: docs/configuring-ingest.asciidoc [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+APM Server default pipelines require you to have the Ingest User Agent Plugin installed.
```
Comment (jalvz) [inline docs/configuring-ingest.asciidoc]: I think this should be lower case: "user agent plugin"

Comment (jalvz) [inline docs/configuring-ingest.asciidoc]: i'd mention what happens if you forgot to install the plugin

File: docs/configuring-ingest.asciidoc [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+Find the default pipeline configuration at `ingest/pipeline/definition.json` of your APM Server
+installation's home directory.
+In case you want to add, change or remove pipelines in Elasticsearch,
+you can simply change the definitions in this file
+and restart your APM Server or run `apm-server setup --pipeline`
+
+By default pipeline registration is disabled,
+you need to set `apm-server.register.ingest.pipeline` to true to enable.
```
Comment (jalvz) [inline docs/configuring-ingest.asciidoc]: i'd remove this:
"you need to set `apm-server.register.ingest.pipeline` to true to enable"

is evident and explained in the config section

File: docs/configuring-ingest.asciidoc [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+See available configuration options for <<register.ingest.pipeline ,pipeline registration>>.
```
Comment (jalvz) [inline docs/configuring-ingest.asciidoc]: Alternative:
"See how to configure <<register.ingest.pipeline ,pipeline registration>>."

File: docs/configuring-ingest.asciidoc [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+[[register-pipelines-manual]]
+[float]
+==== Manually upload pipeline definitions
+For example, let's say that you've defined the following pipeline in a file
```
Comment (jalvz) [inline docs/configuring-ingest.asciidoc]: I'd rephrase as: "consider the following pipeline in a file named `pipeline.json`"

File: docs/configuring-ingest.asciidoc [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+        {
+            "lowercase": {
+                "field": "beat.name"
+            }
+        }
+    ]
+}
+------------------------------------------------------------------------------
+
+To add the pipeline in Elasticsearch, you would run:
```
Comment (jalvz) [inline docs/configuring-ingest.asciidoc]: alternative:
"To register the pipeline, run:"

File: docs/configuring-ingest.asciidoc [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+[source,shell]
+------------------------------------------------------------------------------
+curl -H 'Content-Type: application/json' -XPUT 'http://localhost:9200/_ingest/pipeline/test-pipeline' -d@pipeline.json
+------------------------------------------------------------------------------
+
+This pipeline definition would convert the value of `beat.name` to lowercase before indexing a document.
```
Comment (jalvz) [inline docs/configuring-ingest.asciidoc]: would convert -> converts
before indexing a document -> before indexing each document

File: docs/configuring-ingest.asciidoc [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+
+[[apply-pipelines]]
+[float]
+=== Apply pipelines during data ingestion
+To specify which pipelines to apply during data ingestion,
+you specify the pipeline IDs in the `pipelines` option under `elasticsearch`
```
Comment (jalvz) [inline docs/configuring-ingest.asciidoc]: you specify -> add

i guess we don't need the `{beatname_lc}`? usually we just say APM Server

File: docs/configuration-process.asciidoc [hunk 11]
```diff
@@ -117,6 +117,25 @@ Defaults to `debug/vars`.
 Experimental Metrics endpoint for collecting application metrics.
 Enabled by default.
 
+[[register.ingest.pipeline]]
+[float]
+==== `register.ingest.pipeline`
+A pipeline is a definition of processors registered in Elasticsearch,
```
Comment (jalvz) [inline docs/configuration-process.asciidoc]: Pipelines are already described in `configuring-ingest-node`,  and eg. here: https://www.elastic.co/guide/en/elasticsearch/reference/master/pipeline.html
maybe link to that instead of repeating?

**Comment by simitt** [comment]
Thanks for the thorough review and the suggestions @jalvz , I updated the PR accordingly.
